### PR TITLE
Refactor/flexmatch as matchmaking resource

### PR DESCRIPTION
**Pull request by einarth**

### CODE REVIEW

**Comment by judgeaxl** [comment]
```
Error: This repo is not allowlisted for Atlantis.
```

File: driftbase/api/matchmakers/__init__.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+__matchmakers__ = {"flexmatch"}
```
[the remark this answers is not in the document]
Comment (einarth) [reply driftbase/api/matchmakers/__init__.py]: no, fixed.

File: driftbase/api/matchmakers/__init__.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+
+@bp.route("", endpoint="list")
+class MatchmakersAPI(MethodView):
+
+    def get(self):
+        """
+        Get the available matchmakers
+        """
+        return jsonify(list(__matchmakers__))
```
[the remark this answers is not in the document]
Comment (einarth) [reply driftbase/api/matchmakers/__init__.py]: Rgr, already added one for this guy, will add ones inside the flexmatch module soon

File: driftbase/api/matchmakers/__init__.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+bp = Blueprint("matchmakers", __name__, url_prefix="/matchmakers", description="Discover matchmakers")
+endpoints = Endpoints()
+
+def drift_init_extension(app, api, **kwargs):
+    api.register_blueprint(bp)
+    endpoints.init_app(app)
+
+
+@bp.route("", endpoint="list")
+class MatchmakersAPI(MethodView):
```
[the remark this answers is not in the document]
Comment (einarth) [reply driftbase/api/matchmakers/__init__.py:25]: yup, that makes sense. Deferring that to later though as I'm planning on merging later today in prep for branching off for staging on monday.

[the remark this answers is not in the document]
Comment (judgeaxl) [reply driftbase/api/matchmakers/__init__.py:25]: Fair enough

**Pull request merged**